lib/util/regions.js

```diff
@@ -355,6 +355,24 @@ export function getRegionsFromOverlaps(overlaps, deviceInfo) {
                 fileNames,
             });
 
+        // The content read from device flash memory via JLink is padded.
+        // If the gap between SoftDevice and Application is too small,
+        // then the content read by nrfjprog will be regarded as a single part.
+        // Split them apart if this happens.
+        if (region.name === RegionName.SOFTDEVICE
+            && regionSize - sdRegion.regionSize > deviceInfo.pageSize) {
```

Comment:
Consider adding parenthesis to avoid relying on operator precedence..

Reply:
I think operator precedence is fine here. maybe adding parenthesis will be safer, I will do that.